bitsv/network/transaction.py

```diff
@@ -1,20 +1,19 @@
 
 
 class Transaction:
-    """
-    Representation of a transaction returned from the network.
-    """
+    """Represents a transaction returned from the network."""
 
-    def __init__(self, txid, block, amount_in, amount_out, amount_fee):
+    __slots__ = ('txid', 'amount_in', 'amount_out', 'fee', 'inputs', 'outputs')
+
+    def __init__(self, txid, amount_in, amount_out):
         self.txid = txid
-        self.block = block
+        self.fee = amount_in - amount_out
 
-        if amount_in != amount_out + amount_fee:
-            raise ArithmeticError("the amounts just don't add up!")
+        if amount_in != amount_out + self.fee:
```

Comment:
The changes invalidated this check. Example:
```py
>>> amount_in = 10
>>> amount_out = 15
>>> fee = amount_in - amount_out
>>> fee
-5
>>> amount_in != amount_out + fee
False
```

I suggest moving the check to before the fee calculation and simply using `amount_in - amount_out < 0` as the condition.

Reply:
Oh wow. Testing should have caught this.